src/charm.py

```diff
@@ -125,23 +123,30 @@ def _configure_sdcore_smf(
             self.unit.status = WaitingStatus("Waiting for container to be ready")
             event.defer()
             return
-        if not self._default_database_is_available:
-            self.unit.status = WaitingStatus(
-                "Waiting for default database relation to be available"
-            )
+        if not self._database_is_available:
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for database relation to be available")
             return
         if not self._smf_database_is_available:
+            event.defer()
             self.unit.status = WaitingStatus("Waiting for SMF database relation to be available")
             return
         if not self._nrf_is_available:
+            event.defer()
             self.unit.status = WaitingStatus("Waiting for NRF relation to be available")
             return
-        # TODO: Outsource to separate function for handling database creted event.
-        if not self._config_file_is_written:
-            self._write_config_file(database_url=event.uris.split(",")[0], nrf_url=event.url)  # type: ignore[union-attr] # noqa: E501 TODO: remove.
         if not self._ue_config_file_is_written:
-            self.unit.status = WaitingStatus("Waiting for UE config file to be written")
+            event.defer()
+            self.unit.status = WaitingStatus(
+                f"Waiting for `{UE_CONFIG_FILE}` config file to be pushed to workload container"
+            )
             return
```

Comment:
I want to delete this check because the ue config file must always be there, as it's pushed on install. However, I wanted to ask just in case you've experienced any situation for having to wait to be pushed. If don't I'll remove it.